src/reporting/Market.sol

```diff
@@ -0,0 +1,653 @@
+// Copyright (C) 2015 Forecast Foundation OU, full GPL notice in LICENSE
+
+pragma solidity ^0.4.13;
+
+import 'ROOT/reporting/Branch.sol';
+import 'ROOT/reporting/ReportingToken.sol';
+import 'ROOT/reporting/ReputationToken.sol';
+import 'ROOT/reporting/DisputeBondToken.sol';
+import 'ROOT/reporting/Interfaces.sol';
+import 'ROOT/trading/Cash.sol';
+import 'ROOT/libraries/DelegationTarget.sol';
+import 'ROOT/extensions/MarketFeeCalculator.sol';
+import 'ROOT/factories/MapFactory.sol';
+import 'ROOT/factories/ShareTokenFactory.sol';
+import 'ROOT/factories/ReportingTokenFactory.sol';
+import 'ROOT/factories/DisputeBondTokenFactory.sol';
+import 'ROOT/libraries/Typed.sol';
+import 'ROOT/libraries/Initializable.sol';
+import 'ROOT/libraries/token/ERC20Basic.sol';
+import 'ROOT/libraries/math/SafeMathUint256.sol';
+import 'ROOT/libraries/math/SafeMathInt256.sol';
+
+
+contract Market is DelegationTarget, Typed, Initializable {
+    using SafeMathUint256 for uint256;
+    using SafeMathInt256 for int256;
+
+    enum ReportingState {
+        AUTOMATED,
+        LIMITED,
+        ALL,
+        FORK,
+        FINALIZED
+    }
+
+    ReportingState private reportingState = ReportingState.AUTOMATED;
```

Comment:
Unfortunately, I believe that this won't work correctly.  The most immediate problem is that automated reporting is an optional phase and the market automatically switches from automated reporting phase to limited reporting phase based on `block.timestamp`, not based on a transaction trigger.  Because contracts don't support any kind of alarm clock/trigger system we need to dynamically check what state the contract is in based on what the current `block.timestamp` is.